File: bridge/polyfill/src/document/comment.ts
import { Node, NodeType } from './node';
import { createComment } from './ui-manager';

export class Comment extends Node {
  public nodeName: string = '#comment';
  public data: string = '';

  constructor(data: string) {
    super(NodeType.COMMENT_NODE);
    this.textContent = this.nodeValue = this.data = data;
    createComment(this.targetId, data);
  }
}

File: kraken/lib/src/dom/comment.dart
/*
 * Copyright (C) 2019-present The Kraken authors. All rights reserved.
 */
import 'package:flutter/rendering.dart';
import 'package:kraken/dom.dart';

class Comment extends Node {
  Comment([context]) : super(NodeType.COMMENT_NODE, context);

  @override
  String get nodeName => '#comment';

  @override
  RenderBox? get renderer => null;

  // @TODO: Get data from bridge side.
  String get data => '';

  int get length => data.length;
}
